orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -1502,5 +1502,91 @@ IMPLEMENT_GRADIENT_BUILDER(GetAbsGradient) {
   };
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetMinMaxGradient) {
```

Comment:
Hmm... this gradient involves a lot of small elementwise computation... 
I am wondering if the Max Op has an addition output, i.e. the index of selected input, can help reducing the amount of computation. 
Just like the MaxPool's second output.

Reply:
Those small element wise ops are mainly to handle the case when number of inputs are greater than 2, for number of inputs is 2 (which is the common case, expecially when using PyTorch, whose Min/Max supports only 2 arguments), the code can generate the optimized sub-graph, it's Equal->Cast->Mul for one input, and Not->Cast->Mul for the other one.
If we add the indices output, do we need to start the change from ONNX? And I think the optimized way to use the indices output in backward is to add and write the kernel for Min/MaxGrad Ops, just like MaxPoolGrad. If not, i.e., still use existing Ops, for each input, it will still need Equal->Cast->Mul. The question is, how much perf gain we can get by doing this?

Reply:
I don't want to block this PR's merge, but it seems we need more time to figure out the gradient behavior for more than 2 inputs. 

How about we limit the gradient builder to only support 2 input case? It can ORT_THROW for the other cases?